### PR TITLE
ucs development

### PR DESCRIPTION
**Pull request by jikui**

### CODE REVIEW

File: bigip/common.sh [hunk 1]
```diff
@@ -736,3 +737,144 @@ function get_mgmt_ip_by_name(){
...
+
+    if [ ! -f $2 ];then
+    	echo "UCS file $2 doesn't exist."
+	return 1
+    fi
+
+    echo "Upload ucs file $2 to bigip $1"
+     
+    local SIZE=$(du -b $2 | awk '{ print $1 }')
+    curl -sk -u admin:$BIGIP_ADMIN_PASS --connect-timeout 30 -X POST \
```
Comment (zhaoqin-github) [inline bigip/common.sh:848]: Why not use TIMEOUT var here?

File: bigip/ucs_bigip.sh [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+  "ve-number"
+  "action"
+  "directory"
+)
+
+SHORT_PARAMS="i:n:a:d:h"
+
+OPTS=$(getopt \
+  -o $(printf $SHORT_PARAMS) \
+  -l $(printf "%s:," ${LONG_PARAMS_LIST[@]}) \
```
Comment (zhaoqin-github) [inline bigip/ucs_bigip.sh:105]: We removed ":" of long option here, in order to support long option without parameter. Please see the latest create_bigip.sh code.

File: bigip/common.sh [hunk 3]
```diff
@@ -736,3 +737,144 @@ function get_mgmt_ip_by_name(){
...
+    return 0
+}
+
+#restore the ucs for a bigip
+# $1 bigip mgmt ip
+# $2 file name located in bigip that recover from
+function restore_ucs() {
+    local MGMT_IP=$1
+    echo "restore ucs for host $MGMT_IP"
+    resp=$(curl -sk -u admin:$BIGIP_ADMIN_PASS --connect-timeout 30 -w "\n%{http_code}" -X POST \
```
Comment (zhaoqin-github) [inline bigip/common.sh:868]: Why not use TIMEOUT var here?

File: bigip/common.sh [hunk 4]
```diff
@@ -736,3 +737,144 @@ function get_mgmt_ip_by_name(){
...
+        if [[ $code -ne 200 ]] ; then
+            echo "download ucs last part error. $code"
+	    rm -f "${TMP_FILE}"*
+            return 1
+        fi
+    	cat ${MGMT_IP}.tmp.ucs.part* > $2
+	rm -f "${TMP_FILE}"*
+    elif [[ $code -eq 200 ]] ; then
+        resp=$(curl -H "Content-Type: application/octet-stream" \
+             -sk -u admin:$BIGIP_ADMIN_PASS -w "\n%{http_code}" --connect-timeout 30 -X  GET \
```
Comment (zhaoqin-github) [inline bigip/common.sh:818]: Why not use TIMEOUT var here?

File: bigip/common.sh (hunk 1 above)
Comment (jikui) [reply bigip/common.sh:848]: resolved

File: bigip/ucs_bigip.sh (hunk 2 above)
Comment (jikui) [reply bigip/ucs_bigip.sh:105]: resolved

File: bigip/common.sh (hunk 3 above)
Comment (jikui) [reply bigip/common.sh:868]: resolved

File: bigip/common.sh (hunk 4 above)
Comment (jikui) [reply bigip/common.sh:818]: resolved

File: bigip/ucs_bigip.sh [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+BIGIP_IDX=${BIGIP_IDX:-1}
+BIGIP_NUM=${BIGIP_NUM:-1000}
+UCS_OPERATION=""
+
+export BIGIP_ADMIN_PASS=${BIGIP_ADMIN_PASS:-admin}
+export BIGIP_ROOT_PASS=${BIGIP_ROOT_PASS:-default}
+
+TIMEOUT=30
+LONG_PARAMS_LIST=(
+  "ve-index"
```
Comment (zhaoqin-github) [inline bigip/ucs_bigip.sh:95]: you may need to place ":" here

Comment (pei-jikui) [reply bigip/ucs_bigip.sh]: resolved

Comment (jikui) [reply bigip/ucs_bigip.sh]: resolved

**Pull request merged**